tools/minHash/minHash.xml

```diff
@@ -0,0 +1,196 @@
+block_size = 5
+shingle = 0
+store_value_with_least_significant_bit = 0 
+
+#if $advanced_options.computing_method.method == 'kneighbors':
+
+n_neighbors = $advanced_options.computing_method.k_neighbors
+radius = 1.0
+
+#else: 
+
+n_neighbors = 5
+radius = $advanced_options.computing_method.radius
+
+#end if
+
+
+
+#if bool($advanced_options.prune_index.prune_bool):
+
+prune_inverse_index = $advanced_options.prune_index.prune_value
+
+#if bool($advanced_options.prune_index.prune_index_during_fitting.prune_index_during_fitting_bool):
+
+prune_inverse_index_after_instance = $advanced_options.prune_index.prune_index_during_fitting.prune_inverse_index_after_instance
+
+#else:
+
+prune_inverse_index_after_instance = -1.0
+
+#end if
+
+
+#else:
+
+prune_inverse_index = -1
+prune_inverse_index_after_instance = -1.0
+
+#end if
+
+
+
+#if bool($advanced_options.remove_entries.remove_entries_bool):
+remove_hash_function_with_less_entries_as = $advanced_options.remove_entries.remove_hash_function_with_less_entries_as
+#else: 
+remove_hash_function_with_less_entries_as = -1.0
+#end if
+
+#if bool($advanced_options.shingle.shingle_bool):
+shingle = $advanced_options.shingle.shingle_size
+block_size = $advanced_options.shingle.block_size
+#else:
+shingle = 0
+block_size = 5
+#end if
+
+#if bool($advanced_options.least_significant_bit.least_significant_bit_bool):
+store_value_with_least_significant_bit = $advanced_options.least_significant_bit.store_value_with_least_significant_bit
+#else:
+store_value_with_least_significant_bit = 0
+#end if
+
+minHash = MinHash(n_neighbors = int(n_neighbors), radius = float(radius), fast=bool($advanced_options.fast), 
+                    number_of_hash_functions=int($advanced_options.number_of_hash_functions),
+                    max_bin_size = int($advanced_options.max_bin_size), minimal_blocks_in_common = int($advanced_options.minimal_blocks_in_common),
+                    shingle_size = int(shingle_size), excess_factor = int($advanced_options.excess_factor),
+                    similarity=bool($advanced_options.similarity), number_of_cores=int($advanced_options.number_of_cores),
+                    chunk_size=int($advanced_options.chunk_size), prune_inverse_index=int(prune_inverse_index),
+                    prune_inverse_index_after_instance=float(prune_inverse_index_after_instance), 
+                    remove_hash_function_with_less_entries_as=int(remove_hash_function_with_less_entries_as), 
+                    block_size=int(block_size), shingle=int(shingle),
+                    store_value_with_least_sigificant_bit=int(store_value_with_least_significant_bit))
+dist, indices = minHash.fit_kneighbors(data_matrix)
+mmwrite(open("$dist", 'w+'), dist) 
+mmwrite(open("$indices", 'w+'), indices)
+      ]]>
+    </configfile>
+  </configfiles>
+
+  
+    <inputs>
+        <param name="input_file" type="data" format="mtx" label="Source file" help="Needs to be a sparse matrix from scipy/numpy in csr format."/>
```

Comment:
Use the source Luke :)
https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample
That it should be added I think.